### PR TITLE
Cycle error correction

### PR DESCRIPTION
**Pull request by blogcraft**
**What kind of change does this PR introduce?** (check one with "x")
- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
There is a build problem caused by a dependency cycle that surfaced with #1699 causing to fail when build in prod mode.

<details><summary>Error detaill</summary>

------------------------------------------------------------------------------
Building entry point '@swimlane/ngx-charts'
------------------------------------------------------------------------------
- Compiling with Angular sources in Ivy partial compilation mode.
✖ Compiling with Angular sources in Ivy partial compilation mode.
An unhandled exception occurred: projects/swimlane/ngx-charts/src/lib/pie-chart/pie-grid.component.ts:34:1 - error NG3003: One or more import cycles would need to be created to compile this component, which is not supported by the current compiler configuration.

 34 @Component({
    ~~~~~~~~~~~~
 35   selector: 'ngx-charts-pie-grid',
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
... 
261   }
    ~~~
262 }
    ~

  projects/swimlane/ngx-charts/src/lib/pie-chart/pie-grid-series.component.ts:25:1
     25 @Component({
        ~~~~~~~~~~~~
     26   selector: 'g[ngx-charts-pie-grid-series]',
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ... 
    118   }
        ~~~
    119 }
        ~
    The component 'PieGridSeriesComponent' is used in the template but importing it would create a cycle: /home/runner/work/ngx-charts/ngx-charts/projects/swimlane/ngx-charts/src/lib/pie-chart/pie-grid.component.ts -> /home/runner/work/ngx-charts/ngx-charts/projects/swimlane/ngx-charts/src/lib/pie-chart/pie-grid-series.component.ts -> /home/runner/work/ngx-charts/ngx-charts/projects/swimlane/ngx-charts/src/lib/pie-chart/pie-grid.component.ts

</details>
**What is the new behavior?**
Now the Lib compiles correctly.


**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:

### CODE REVIEW

**Comment by NashIlli** [comment]
+1

**Comment by joeveiga** [comment]
@blogcraft Thanks for the PR! It seems like the issue is caused by the import of the model. If that's the case I'd much rather move that `PieGridData` interface somewhere more suitable, perhaps `lib/models/chart-data.model.ts`? As opposed to just moving it to the other component ts file.

**Review by joeveiga — CHANGES_REQUESTED** [review]
It'd be better to move this interface to `models/` folder, IMO.

**Comment by blogcraft** [comment]
@joeveiga I moved the interfaces to the model folder inside chart-data.model.ts

**Comment by blogcraft** [comment]
I believe that This might fail the tests during release because of prettier.

Should I do an other commit after runing `npm run prettier`?

File: projects/swimlane/ngx-charts/src/lib/models/chart-data.model.ts [hunk 1]
```diff
@@ -36,6 +36,24 @@ export interface PieGridDataItem extends DataItem {
...
 
+export interface PieGridData {
+  data: PieGridDataItem;
+  height: number;
+  width: number;
+  x: number;
+  y: number;
+}
+
+export interface PieArc {
```
Comment (joeveiga) [inline projects/swimlane/ngx-charts/src/lib/models/chart-data.model.ts:47]: @blogcraft Was this `PieArc` interface also giving you problems? If not, I'd leave it where it was (at least for now) as it appears it is not imported anywhere else.

**Review by joeveiga — CHANGES_REQUESTED** [review]

File: projects/swimlane/ngx-charts/src/lib/models/chart-data.model.ts (hunk 1 above)
Comment (blogcraft) [reply projects/swimlane/ngx-charts/src/lib/models/chart-data.model.ts]: Pie Arc needs to be moved too, if not the Cycle error persists:

pie-grid.component.ts -> pie-grid-series.component.ts -> pie-grid.component.ts

**Comment by blogcraft** [comment]
> > I believe that This might fail the tests during release because of prettier.
> > Should I do an other commit after runing `npm run prettier`?
> 
> You can run a `npm test` to check everything is ok :). But I don't expect any issues as it is a fairly small change. I'd also ask to update the changelog (under 'unreleased') with the fix if it is not much trouble? Thanks!

I executed `npm test` and this is the output:
```
[warn] ...... (many warnings)
[warn] Code style issues found in the above file(s). Forgot to run Prettier?
⠼ Generating browser application bundles (phase: building)...ERROR: "prettier:ci" exited with 1.
```
The warning list is big, and if I execute `npm prettier`, many files are modified (no code is changed, but it seems that the EOF is changed from CRLF to LF).

**Comment by joeveiga** [comment]
> The warning list is big, and if I execute `npm prettier`, many files are modified (no code is changed, but it seems that the EOF is changed from CRLF to LF).
Hmmm.. not sure what that's about.. Alright, let me open a quick PR with the fix over here and I'll link this one.

**Comment by joeveiga** [comment]
Closed in favor of #1722. Thanks @blogcraft !

**Pull request closed without merging**